### ISSUE TITLE
In-memory cross-project reference not working for project using a generative type provider

### ISSUE THREAD

**Issue by jkone27** [dotnet/fsharp#11771]
When I use VS Build has 0 errors, but VS ide says there is errors.. this repo should allow to reproduce https://github.com/jkone27/test-broken

### Workaround

Tools -> Options -> Text Editor -> F# -> Performance -> Turn off "Enable in-memory cross project references"

### Details

I am on win10 and vs2019 (and vs2020 also affected)

e..g in another repo

![image](https://user-images.githubusercontent.com/10419217/124271469-c7697f80-db3d-11eb-8bad-664f951b7b3b.png)

**Comment by dsyme** [comment]
Thank you for the report and repro

**Comment by dsyme** [comment]
I can confirm this is a bug

Workaround: Tools -> Options -> Text Editor -> F# -> Performance -> Turn off "Enable in-memory cross project references"

**Comment by dsyme** [comment]
@jkone27 Did you only experience this with repos that involve the Swagger type provider?  If you have any other repros please include them

**Comment by dsyme** [comment]
@TIHan I'm recalling the details now.  In brief, in-memory cross-project references simply don't work for projects involving generative type providers.  However they should be implicitly disabled, that's what this PR was about: https://github.com/dotnet/fsharp/pull/3236

I suspect this implicit disablement is no longer functioning.  The relevant code is still there:

```fsharp
                        if tcState.CreatesGeneratedProvidedTypes || hasTypeProviderAssemblyAttrib then
                            None
                        else
                            Some  (RawFSharpAssemblyDataBackedByLanguageService (tcConfig, tcGlobals, generatedCcu, outfile, topAttrs, assemblyName, ilAssemRef) :> IRawFSharpAssemblyData)
```

Perhaps if this kicks in the project assembly reference is now being completely dropped on the floor?

**Comment by dsyme** [comment]
This looks like the problem:
```
        // If we have a project reference but did not get any valid contents,
        //     just return None and do not attempt to read elsewhere.
        if contentsOpt.IsNone && r.ProjectReference.IsSome then
            return None
        else
```

It is valid to get `None` in this case, and we should revert to the on-disk DLL if it exists, as a known limitation of the tooling.

**Comment by dsyme** [comment]
Fixed by #11791
